src/toil/batchSystems/mesos/executor.py

```diff
@@ -124,7 +124,7 @@ def runJob(job):
                 job.userScript.register()
             log.debug("Invoking command: '%s'", job.command)
             with self.popenLock:
-                return subprocess.Popen(job.command, shell=True)
+                return subprocess.Popen(job.command, shell=True, env=job.environment)
```

Comment:
Here you need to mix the current environment with job.environment. Otherwise, the current environment would be lost and the only variables requested via setEnv would be present. But we want the current environment variables *plus* the ones requested via setEnv.